SwiftRustIntegrationTestRunner/test-release-fails/README.md

```diff
@@ -0,0 +1,7 @@
+# Test Release Builds Fail
```

Comment:
Let's document:

- Why it would infer that they were dead code (as in, mention `func` vs. `public func`)

-  show an example of a problematic `cdecl` function. Explain that we now prepend `public`.
    ```swift
    @_cdecl("__swift_bridge__$add") 
    func __swift_bridge__add (ptr: UnsafeMutableRawPointer) { /* .. */ }
    ```
    - This will help orient the reader and give them a better understanding of what we're testing

- How the test works. How are we accomplishing what you've explained here? Maybe explain how the test works step by step.

- How to run the test.

- Mention that this test gets ran during `test-swift-rust-integration.sh`

All of this will help future maintainers understand what's going on and be in a better position to make changes/improvements to it.